addons/mail/models/mail_message.py

```diff
@@ -958,13 +958,26 @@ def message_format(self, format_reply=True, msg_vals=None):
                     'is_discussion': False # only if the message is a discussion (subtype == discussion)
                     'is_notification': False # only if the message is a note but is a notification aka not linked to a document like assignation
                     'parentMessage': {...}, # formatted message that this message is a reply to. Only present if format_reply is True
+                    'user_follower_id': Follower id if user is following the related record otherwise False
                 }
         """
         vals_list = self._message_format(self._get_message_format_fields(), format_reply=format_reply)
 
         com_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment')
         note_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_note')
-
+        records_followed = self.env['mail.followers'].sudo().search(
+            expression.AND([
+                expression.OR([
+                    [('res_model', '=', model), ('res_id', 'in', list({value['res_id'] for value in values}))]
+                    for model, values in
+                    misc_groupby((vals for vals in vals_list if vals.get("model") and vals.get("res_id")),
+                                 key=lambda r: r["model"])
+                ]),
+                [('partner_id', '=', self.env.user.partner_id.id)],
+            ])
+        )
+        followed_refs = {(res['res_model'], res['res_id']): res['id']
```

Comment:
Thanks for the suggestion. Actually, I have changed that part with the new discuss implementation because message_format was not always called by the partner that displays the messages while the message were transmitted to the user with the bus. So we couldn’t have information only related to one follower in the message formatted. More details in the commit messages.